samtranslator/model/eventsources/pull.py

```diff
@@ -110,6 +110,9 @@ def to_cloudformation(self, **kwargs):
 
         destination_config_policy = None
         if self.DestinationConfig:
+            if self.DestinationConfig.get("OnFailure") is None:
+                raise InvalidEventException(self.logical_id, "'OnFailure' is a required field for 'DestinationConfig'")
```

Comment:
The [docs](https://docs.aws.amazon.com/AWSCloudFormation/latest/UserGuide/aws-properties-lambda-eventsourcemapping-destinationconfig.html) for `DestinationConfig` show that `OnFailure` is not a required field.  I'm not sure why someone would create `DestinationConfig` with its only possible field so I think the docs are incorrect.  Either way, we should make the code and the docs consistent.

Reply:
There's a discrepancy between doc and code. In addition to the one you mentioned above, the doc is also missing to show `Type` as required under `OnFailure`. I'll request a doc update to address these inconsistencies.